### PR TITLE
Add SDXL model

### PR DESCRIPTION
**Pull request by zhanghuiyao**
Upload SDXL model:
1. text-to-image generative
2. lora finetune

### CODE REVIEW

**Review by DongyaoZhu — COMMENTED** [review]
For model weight conversion, I would be consistent across examples:
scripts/convert -> tools/model_conversion

**Review by DongyaoZhu — COMMENTED** [review]
merge readmes?

**Comment by zhanghuiyao** [comment]
> merge readmes?

merge getting_started.md to readme.md? may clearer to separate?

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune.yaml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                lossconfig:
+                    target: mindspore.nn.Identity
+
+
+        # Module for Train
+        scheduler_config:
+            target: gm.lr_scheduler.LambdaLinearScheduler
+            params:
+                warm_up_steps: [ 10000 ]
+                cycle_lengths: [ 10000000000000 ]
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune.yaml:108]: use 1e13 for clarity?

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora.yaml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+                scaling_config:
+                    target: gm.modules.diffusionmodules.denoiser_scaling.EpsScaling
+                discretization_config:
+                    target: gm.modules.diffusionmodules.discretizer.LegacyDDPMDiscretization
+
+        network_config:
+            target: gm.modules.diffusionmodules.openaimodel.UNetModel_lora
+            params:
+                lora_dim: 4                 # for LoRA
+                lora_alpha: 32              # for LoRA
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora.yaml:24]: why is alpha so large? typically it is 1 referring to diffusers or cloneofsimo/lora.

File: examples/stable_diffusion_xl/gm/modules/lora/layers.py [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        LoRALayer.__init__(self, r=r, lora_alpha=lora_alpha, lora_dropout=lora_dropout,
+                           merge_weights=merge_weights)
+
+        # Actual trainable parameters
+        # self.lora_A = Parameter(Tensor(np.zeros((r, in_features)), self.weight.dtype))
+        # self.lora_B = Parameter(Tensor(np.zeros((out_features, r)), self.weight.dtype))
+        self.lora_A = nn.Dense(in_features, r, has_bias=False)
+        self.lora_B = nn.Dense(r, out_features, has_bias=False)
+
+        self.scaling = self.lora_alpha / self.r
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/gm/modules/lora/layers.py:60]: With default lora_alpha=1.0, ·scaling· is 0.25. But actually, we expect scaling = 1.0 by default (referring to diffusion lora).  We may set `lora_alpha: int = None,`, and set scaling to 1. if lora_alpha is None

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora_gpu.yaml [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+                scaling_config:
+                    target: gm.modules.diffusionmodules.denoiser_scaling.EpsScaling
+                discretization_config:
+                    target: gm.modules.diffusionmodules.discretizer.LegacyDDPMDiscretization
+
+        network_config:
+            target: gm.modules.diffusionmodules.openaimodel.UNetModel_lora
+            params:
+                lora_dim: 4                 # for LoRA
+                lora_alpha: 32              # for LoRA
```
Comment (SamitHuang) [inline examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora_gpu.yaml:24]: same here

**Review by SamitHuang — COMMENTED** [review]
Still reviewing...

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora.yaml (hunk 2 above)
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora.yaml:24]: 32 is reference from gpt-2 lora. The experiment seems to have similar effects with alpha 4 and 32, currently modified to the same configuration as in diffuser, alpha -> 4(scale 1).

File: examples/stable_diffusion_xl/gm/modules/lora/layers.py (hunk 3 above)
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/gm/modules/lora/layers.py:60]: fixed

File: examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora_gpu.yaml (hunk 4 above)
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/configs/training/sd_xl_base_finetune_lora_gpu.yaml:24]: lora alpha: 32 -> 4

File: examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        sigma = ops.cast(sigma, ms.float32)
+        sigma = self.possibly_quantize_sigma(sigma)
+        sigma_shape = sigma.shape
+        sigma = append_dims(sigma, input_dim)
+        c_skip, c_out, c_in, c_noise = self.scaling(sigma)
+        c_noise = self.possibly_quantize_c_noise(c_noise.reshape(sigma_shape))
+
+        return c_skip, c_out, c_in, c_noise
+
+    # zhy_test
```
[the remark this answers is not in the document]
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/gm/modules/diffusionmodules/denoiser.py:34]: fixed

File: examples/stable_diffusion_xl/gm/util/util.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+
+def seed_everything(seed):
+    random.seed(seed)
+    np.random.seed(seed)
+    ms.set_seed(seed)
+
+
+#### for mix precision
+def auto_mixed_precision(network, amp_level="O0"):
```
[the remark this answers is not in the document]
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/gm/util/util.py:77]: The default interface cannot meet the requirements

File: examples/stable_diffusion_xl/GETTING_STARTED.md [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+
+- [pokemon-blip-caption dataset](https://openi.pcl.ac.cn/jasonhuang/mindone/datasets), containing 833 pokemon-style images with BLIP-generated captions.
+- [Chinese-art blip caption dataset](https://openi.pcl.ac.cn/jasonhuang/mindone/datasets), containing 100 chinese art-style images with BLIP-generated captions.
+
+To use them, please download `pokemon_blip.zip` or `chinese_art_blip.zip` from the [openi dataset website](https://openi.pcl.ac.cn/jasonhuang/mindone/datasets). Then unzip.
+
+</details>
+
+
+## Inference
```
[the remark this answers is not in the document]
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/GETTING_STARTED.md:53]: Added

File: examples/stable_diffusion_xl/gm/helpers.py
[the remark this answers is not in the document]
Comment (zhanghuiyao) [reply examples/stable_diffusion_xl/gm/helpers.py:1]: done

**Review by SamitHuang — APPROVED** [review]

**Review by DongyaoZhu — APPROVED** [review]
minor revision on English writing (grammar, etc.), can wait for later

**Pull request merged**